Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt

```diff
@@ -49,52 +49,57 @@ class ConfidenceFeatureProvider private constructor(
         }
     }
 
+    private fun startListeningForContext() {
+        coroutineScope.launch {
+            confidence.contextChanges
+                .collect {
+                    resolve(InitialisationStrategy.FetchAndActivate)
+                }
+        }
+    }
+
     override fun initialize(initialContext: EvaluationContext?) {
         initialContext?.let {
-            internalInitialize(
-                initialContext,
-                initialisationStrategy
-            )
+            // refresh cache with the last stored data
+            storage.read().let(providerCache::refresh)
+            if (initialisationStrategy == InitialisationStrategy.ActivateAndFetchAsync) {
+                eventHandler.publish(OpenFeatureEvents.ProviderReady)
+            }
+
+            coroutineScope.launch(networkExceptionHandler) {
+                val context = initialContext.toConfidenceContext()
+                confidence.putContext(context.map)
```

Comment:
In Swift we rather update the context in Confidence directly, and then read from it: https://github.com/spotify/confidence-openfeature-provider-swift/blob/0d1cefdeb766a3d24c7d05be5f834d8855f271f3/Sources/ConfidenceProvider/ConfidenceFeatureProvider.swift#L87

I don't think this is necessarily a blocker, but maintaining a more aligned implementation will make things easier for us

Reply:
> we rather update the context in Confidence directly, and then read from it:
we do the same here i suppose, could you explain more what's the difference?

Reply:
We do the same, you are right